src/app/public/modules/country-field/country-field.component.ts

```diff
@@ -0,0 +1,295 @@
+import {
+  ChangeDetectionStrategy,
+  ChangeDetectorRef,
+  Component,
+  ElementRef,
+  EventEmitter,
+  forwardRef,
+  Input,
+  OnDestroy,
+  OnInit,
+  Output,
+  ViewChild
+} from '@angular/core';
+
+import {
+  AbstractControl,
+  ControlValueAccessor,
+  FormBuilder,
+  FormControl,
+  FormGroup,
+  NG_VALUE_ACCESSOR,
+  ValidationErrors
+} from '@angular/forms';
+
+import {
+  SkyAppWindowRef
+} from '@skyux/core';
+
+import 'intl-tel-input';
+
+import {
+  fromEvent,
+  Subject
+} from 'rxjs';
+
+import {
+  SkyAutocompleteInputDirective,
+  SkyAutocompleteSelectionChange
+} from '../autocomplete';
+
+import {
+  SkyCountryFieldCountry
+} from './types';
+
+@Component({
+  selector: 'sky-country-field',
+  templateUrl: './country-field.component.html',
+  styleUrls: ['./country-field.component.scss'],
+  providers: [{
+    provide: NG_VALUE_ACCESSOR,
+    /* tslint:disable-next-line:no-forward-ref */
+    useExisting: forwardRef(() => SkyCountryFieldComponent),
+    multi: true
+  }],
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class SkyCountryFieldComponent implements ControlValueAccessor, OnDestroy, OnInit {
+
+  /**
+   * Specifies the [International Organization for Standardization Alpha 2](https://www.nationsonline.org/oneworld/country_code_list.htm)
+   * country code for the default country.
+   * This default country will show up at the top of search results when applicable.
+   * @default us
+   */
+  @Input()
+  public set defaultCountry(value: string) {
+    if (value !== this._defaultCountry) {
+      this._defaultCountry = value.toLowerCase();
+
+      this.defaultCountryData = this.countries
+        .find(country => country.iso2 === this._defaultCountry);
+
+      this.sortCountriesWithSelectedAndDefault(this.selectedCountry);
+    }
+  }
+
+  public get defaultCountry(): string {
+    return this._defaultCountry;
+  }
+
+  /**
+   * Indicates whether to disable the country field.
+   */
+  @Input()
+  public disabled: boolean = false;
+
+  /**
+   * Emits when the selected country changes.
```

Comment:
I think "Emits" should be "Fires." Either that, or we should state what the event emits, i.e. `Emits a SkyCountryFieldCountry object when the selected country changes.`

Reply:
I went with "Fires" since the new docs tools outputs the type beside the description and that seemed to be the standard put in place in the popover docs